app/src/main/java/com/android/template/ui/home/HomeActivity.kt

```diff
@@ -93,4 +125,25 @@ class HomeActivity : BaseActivity(), HomeContract.View {
             else -> super.onOptionsItemSelected(item)
         }
     }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        if(requestCode == authUtils.getRequestCode()) {
+            handleSignInResponse(resultCode, data)
+        }
+    }
+
+    @SuppressLint("RestrictedApi") // Suppress for line 138 where user name is accessed from response which is 2 different APIs, probably nothing to worry about.
+    private fun handleSignInResponse(resultCode: Int, data: Intent?) {
+        val response: IdpResponse = IdpResponse.fromResultIntent(data) ?: return
+
+        if(Activity.RESULT_OK == resultCode) {
+            // TODO: do something with the response
+            toastUtil.showShortToast("Welcome, ${response.user.name}")
```

Comment:
Pretpostavljam da je ovo privremeno. SAda bi bila dobra ideja da tu pozoveš metodu na prezenterom `presenter.userAuthenticated()` i onda iz use-caseva pokupi stvari kaj ti treba iz Firebase-a

Reply:
da, to je sljedeći TODO